drivers/adc/adc_npcx.c

```diff
@@ -64,6 +68,17 @@ struct adc_npcx_data {
 	uint16_t *repeat_buffer;
 	/* end pointer of buffer to ensure enough space for storing ADC data. */
 	uint16_t *buf_end;
+#if IS_ENABLED(CONFIG_ADC_THRESHOLD_IRQ)
```

Comment:
IS_ENABLED should't be combined with preprocessor guards.

Use it like `if (IS_ENABLED(...)) { ... }`